tmt/steps/prepare/ansible.py

```diff
@@ -1,9 +1,29 @@
+import sys
+from typing import Any, List, Optional
+
 import click
 
 import tmt
-
-
-class PrepareAnsible(tmt.steps.prepare.PreparePlugin):
+import tmt.utils
+from tmt.steps.provision import Guest
+
+if sys.version_info >= (3, 8):
+    from typing import TypedDict
+else:
+    from typing_extensions import TypedDict
```

Comment:
Just a quick thought here: Seems we will be using these quite frequently, so what about importing these in `tmt.utils` just once and then taking `TypedDict` and other python3.8+ friends from there?

Reply:
I'm going to do exactly that. I added a `_typing.py` in https://github.com/teemtee/tmt/pull/1178/files#diff-abf016eefb48b622c11ce73a4c723c0ab2758b9fb7be480b873cedb2882d5d1b, when there were not so many users yet, but with the influx of type annotations, I could make this a standalone patch.

Would you like me to do so before `ansible.py` changes, or would it be acceptable after? If that would be fine, the burden of fixing of files would be on me, no additional rebases for poor @guy9050 :)

Reply:
> I'm going to do exactly that. I added a `_typing.py` in...

Ah, nice!

> Would you like me to do so before ansible.py changes, or would it be acceptable after?

I'd be slightly inclined to merge the `_typing` sooner than later. But whatever you prefer more.

Reply:
That's fine, I like cleanup patches. I'll make a PR out of it - my bigger PR needs a rebase anyway, won't be a major issue.

Reply:
@psss bad news: to move these imports into a dedicated module, mypy must be allowed to follow imports. This is now disabled, to avoid mypy reporting issues in modules that were not annotated yet. It is possible to soften this rule, to follow imports but report no issues in imported files, but that setting will still report the use of non-annotated code in the annotated files. For example, when checking `tmt.steps.provision.artemis`, mypy would follow imports of `tmt`, but `artemis.py` is not allowed to use `tmt.steps.Method` class because it'd be using untyped code in typed context.

I tried to enable the softer setting (`follow_imports = silent`), but the number of issues is still too high (e.g. all calls to `self.get()` and so on). If you won't mind, I'd leave the addition of `_typing.py` to a later stage, when more of tmt's codebase would gain annotations. With a `silent` setting, the number of issues should lower considerably.

Reply:
Understood, make sense.